fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -1563,59 +1563,120 @@ public void replayAddPartition(PartitionPersistInfo info) throws DdlException {
+                    throw new DdlException("Batch deletion of partitions only support range partition tables.");
+                }
+                RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionInfo;
+
+                List<Column> partitionColumns = rangePartitionInfo.getPartitionColumns();
+                if (partitionColumns.size() != 1) {
+                    throw new DdlException("Alter batch drop partition only support single range column.");
+                }
+
+                Column firstPartitionColumn = partitionColumns.get(0);
+                MultiRangePartitionDesc multiRangePartitionDesc = (MultiRangePartitionDesc) partitionDesc;
+                boolean isAutoPartitionTable = false;
+                if (partitionInfo.getType() == PartitionType.EXPR_RANGE) {
+                    isAutoPartitionTable = true;
+                    ExpressionRangePartitionInfo exprRangePartitionInfo = (ExpressionRangePartitionInfo) partitionInfo;
+                    Expr expr = exprRangePartitionInfo.getPartitionExprs().get(0);
+                    if (expr instanceof FunctionCallExpr) {
+                        FunctionCallExpr functionCallExpr = (FunctionCallExpr) expr;
+                        checkAutoPartitionTableLimit(functionCallExpr, multiRangePartitionDesc);
                     }
                 }
+                Map<String, String> properties = clause.getProperties();
+                if (properties == null) {
+                    properties = Maps.newHashMap();
+                }
+                if (tableProperties != null && tableProperties.containsKey(DynamicPartitionProperty.START_DAY_OF_WEEK)) {
+                    properties.put(DynamicPartitionProperty.START_DAY_OF_WEEK,
+                            tableProperties.get(DynamicPartitionProperty.START_DAY_OF_WEEK));
+                }
+                PartitionConvertContext context = new PartitionConvertContext();
+                context.setAutoPartitionTable(isAutoPartitionTable);
+                context.setFirstPartitionColumnType(firstPartitionColumn.getType());
+                context.setProperties(properties);
+                context.setTempPartition(clause.isTempPartition());
+                List<SingleRangePartitionDesc> singleRangePartitionDescs = multiRangePartitionDesc.convertToSingle(context);
+                partitionNames.addAll(singleRangePartitionDescs.stream()
+                        .map(SingleRangePartitionDesc::getPartitionName).collect(Collectors.toList()));
             }
-            Range<PartitionKey> partitionRange = null;
-            if (partitionInfo instanceof RangePartitionInfo && partition != null) {
-                partitionRange = ((RangePartitionInfo) partitionInfo).getRange(partition.getId());
+        } else {
+            partitionNames.add(clause.getPartitionName());
+        }
+        List<String> existPartition = Lists.newArrayList();
```

Comment:
It would be better to add `s` for plural form.

Reply:
ok